### PR TITLE
feat(jenkins): add cicd_scope_id to cicd_tasks and cicd_pipelines

### PR DESCRIPTION
**Pull request by warren830**
# Summary

In this pr, we did the following changes:
1. add cicd_scope_id to cicd_tasks and cicd_pipelines
2. let CicdScope implement core.Scope
3. change the struct of jenkinsJob
4. set value of cicd_tasks and cicd_pipelines when converting 

Also as jenkinsBuild and jenkinsStages don't have jenkinsJob.fullName, we add a field `Job        *models.JenkinsJob` to 
`JenkinsTaskData`. And then we can directly use it when generating cicd_scope_id for cicd_tasks and cicd_pipelines

### Does this close any open issues?
Relate to #3699 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

File: plugins/jenkins/e2e/jobs_test.go [hunk 1]
```diff
@@ -62,4 +64,24 @@ func TestJenkinsJobsDataFlow(t *testing.T) {
 			"primary_view",
 		),
 	)
+
+	// verify extraction
+	dataflowTester.FlushTabler(&devops.CicdScope{})
+	dataflowTester.Subtask(tasks.ExtractApiJobsMeta, taskData)
```
Comment (likyh) [inline plugins/jenkins/e2e/jobs_test.go:70]: why duplicate this?

File: plugins/jenkins/models/migrationscripts/20221110_change_index_of_job_path.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	return "_tool_jenkins_jobs"
+}
+
+func (script *changeIndexOfJobPath) Up(basicRes core.BasicRes) errors.Error {
+	db := basicRes.GetDal()
+	err := db.RenameTable("_tool_jenkins_jobs", "tmp_tool_jenkins_jobs")
+	if err != nil {
+		return err
+	}
+	err = db.AutoMigrate(&jenkinsJob20221108{})
```
Comment (likyh) [inline plugins/jenkins/models/migrationscripts/20221110_change_index_of_job_path.go:71]: why migrate model not in this file?

File: plugins/dora/e2e/snapshot_tables/lake_cicd_tasks.csv [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
-github:GithubJob:1:8269729327,xx_deploy_xx,github:GithubRun:1:3022301001,FAILURE,DONE,,PRODUCTION,82,2022-09-09T12:11:22.000+00:00,2022-09-09T12:12:44.000+00:00
-github:GithubJob:1:8269729550,build_deployment,github:GithubRun:1:3022301007,SUCCESS,DONE,,PRODUCTION,6,2022-09-09T12:12:02.000+00:00,2022-09-09T12:12:08.000+00:00
-github:GithubJob:1:8269729553,build_deploy,github:GithubRun:1:3022300997,SUCCESS,DONE,,PRODUCTION,57,2022-09-09T12:12:52.000+00:00,2022-09-09T12:13:49.000+00:00
-github:GithubJob:1:8269729821,deployment,github:GithubRun:1:3022301033,SUCCESS,DONE,,PRODUCTION,991,2022-09-09T12:13:58.000+00:00,2022-09-09T12:30:29.000+00:00
-github:GithubJob:1:8269872730,deploy,github:GithubRun:1:3022348786,SUCCESS,DONE,,PRODUCTION,1006,2022-09-09T12:21:30.000+00:00,2022-09-09T12:38:16.000+00:00
+id,name,pipeline_id,result,status,type,environment,duration_sec,started_date,finished_date,cicd_scope_id
+github:GithubJob:1:7773010057,build_deploy,github:GithubRun:1:3022300997,SUCCESS,DONE,,PRODUCTION,57,2022-09-09T12:12:52.000+00:00,2022-09-09T12:13:49.000+00:00,
```
Comment (likyh) [inline plugins/dora/e2e/snapshot_tables/lake_cicd_tasks.csv:2]: why `cicd_scope_id` is empty here? not extract this column?

Comment (likyh) [reply plugins/dora/e2e/snapshot_tables/lake_cicd_tasks.csv:2]: Does Dora need filter by cicd_scope_id ?

Comment (warren830) [reply plugins/dora/e2e/snapshot_tables/lake_cicd_tasks.csv:2]: because we haven't changed dora yet

File: plugins/jenkins/e2e/jobs_test.go (hunk 1 above)
Comment (warren830) [reply plugins/jenkins/e2e/jobs_test.go:70]: fixed

File: plugins/jenkins/models/migrationscripts/20221110_change_index_of_job_path.go (hunk 2 above)
Comment (warren830) [reply plugins/jenkins/models/migrationscripts/20221110_change_index_of_job_path.go:71]: fixed

**Review by likyh — APPROVED** [review]
LGTM, this pr add model `CicdScope`, add path to primaryKey for jenkinsJob, and convert jenkinsJob to `CicdScope`.

Other `CicdScope` maybe will be converted on another PR.

**Pull request merged**